modules/programs/nh.nix

```
@@ -0,0 +1,110 @@
+# Based off: https://github.com/NixOS/nixpkgs/blob/master/nixos/modules/programs/nh.nix
+# When making changes please try to keep it in sync.
+{ config
+, lib
+, pkgs
+, ...
+}:
+
+with lib;
+
+let
+  cfg = config.programs.nh;
+in
+{
+  meta.maintainers = [
+    maintainers.alanpearce or "alanpearce"
+  ];
+
+  imports = [
+    (mkRemovedOptionModule [ "programs" "nh" "clean" "dates" ] "Use `programs.nh.clean.interval` instead.")
+  ];
+
+  options.programs.nh = {
+    enable = mkEnableOption "nh, yet another Nix CLI helper";
+
+    package = mkPackageOption pkgs "nh" { };
+
+    flake = mkOption {
+      type = types.nullOr types.path;
```

See https://github.com/nix-community/home-manager/pull/6468 and https://github.com/nix-community/home-manager/pull/6898 (allows flake specs like "github:owner/repo")
```suggestion
      type = with types; nullOr (either singleLineStr path);
```